src/client/lazy-app/Compress/index.tsx

```diff
@@ -366,14 +375,30 @@ export default class Compress extends Component<Props, State> {
   }
 
   componentWillUnmount(): void {
-    updateDocumentTitle();
+    updateDocumentTitle({ loading: false });
     this.mainAbortController.abort();
     for (const controller of this.sideAbortControllers) {
       controller.abort();
     }
   }
 
   componentDidUpdate(prevProps: Props, prevState: State): void {
+    const wasLoading =
+      prevState.loading ||
+      prevState.sides[0].loading ||
+      prevState.sides[1].loading;
+    const isLoading =
+      this.state.loading ||
+      this.state.sides[0].loading ||
+      this.state.sides[1].loading;
+    const sourceChanged = prevState.source !== this.state.source;
+    if (wasLoading !== isLoading || sourceChanged) {
+      updateDocumentTitle({
+        loading: isLoading,
+        filename:
+          (isLoading ? loadingIndicator : '') + this.state.source?.file.name,
+      });
```

Comment:
As above, the loading indicator isn't part of the file name.

```suggestion
      updateDocumentTitle({
        loading: isLoading,
        filename: this.state.source?.file.name,
      });
```